lib/nve/nve-main-page.dart

```diff
@@ -77,21 +79,60 @@ class Position with ChangeNotifier {
       );
     }
   }
+
+  class MyStatefulBluePage extends StatefulWidget {
+    MyStatefulBluePage({Key key}) : super(key: key);
   
-  class BluePage extends StatelessWidget {
+    @override
+    BluePage createState() => BluePage();
+  }
+
+  class BluePage extends State<MyStatefulBluePage> {
+    double _distance = 0;
+
+    Future<void> getDistance() async {
+      _locationData = await location.getLocation();
+      _distance = await geo.distanceBetween(_locationData.latitude, _locationData.longitude, 
+                                            Provider.of<Position>(context, listen: false).latitude, 
```

Comment:
This method throws exceptions when the start location is not set yet. Please fix it